File: src/main/java/cn/jgayb/fenixplugin/Fenixs.java
package cn.jgayb.fenixplugin;

import com.intellij.util.xml.*;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * @author yanglin
 */
public interface Fenixs extends DomElement {

    @NotNull
    @SubTagList("fenix")
    List<IdDomElement> getDaoElements();

    @Required
    @NameValue
    @NotNull
    @Attribute("namespace")
    GenericAttributeValue<String> getNamespace();


}

File: src/main/java/cn/jgayb/fenixplugin/Fenixs.kt
package cn.jgayb.fenixplugin

import com.intellij.psi.PsiClass
import com.intellij.util.xml.*

interface Fenixs : DomElement {

    @SubTagList("fenix")
    fun getDaoElements(): List<IdDomElement?>

    @Required
    @NameValue
    @Attribute("namespace")
    fun getNamespace(): GenericAttributeValue<String?>

}

interface IdDomElement : DomElement {
    @get:Attribute("id")
    @get:NameValue
    @get:Required
    val id: GenericAttributeValue<String?>?

    @get:Convert(AliasConverter::class)
    @get:Attribute("resultType")
    val resultType: GenericAttributeValue<PsiClass?>

    fun setValue(content: String?)
}
